tests/unit/credentials_test.py

```diff
@@ -101,6 +101,26 @@ def test_chain_retrieve(self):
         self.assertEqual(creds.secret_key, "secret_aws")
         self.assertEqual(creds.session_token, "token_aws")
 
+    def test_chain_retrieve_second(self):
```

Comment:
Please name the method appropriately.

Reply:
It is indeed appropriate, since we set two providers and we expect to get the second one.